team-telnyx/telnyx-client-examples
# Add agent to conference call

Allows agent on a call to add another available agent.

This includes a major refactor of the call flow to use [conference commands](https://developers.telnyx.com/docs/api/v2/call-control/Conference-Commands). You may want to read the entire [calls.controller](https://github.com/team-telnyx/telnyx-client-examples/blob/d7616a89f404bc470535d51b1f2c9448651830f7/call-center/server/controllers/calls.controller.ts) file top to bottom to check for readability.

## ✋ Manual testing

1. Start the `call-center` app
2. Open up two browser incognito windows and login as as an agent to both clients
3. Dial your call center phone number from another device (or soft phone) once both agents are marked as "Available". Verify that you hear hold music. Verify that only one of the clients rings (fig. 1)
4. Answer the call. Verify that the hold music stops (it may take a second or two) and that you can hear audio on both sides
5. Click "Add" next to the other agents' name. Verify that this rings the other client (fig. 2)
6. Answer as the second agent. Verify that you hear audio on all sides. Verify both clients appear "Busy" (fig. 3)
7. Hang up the call. Call should not hang up for other legs (we may want to rework this logic later)

## 🦊 Browser testing

### Desktop

### Mobile

## 📸 Screenshots

**Figure 1: 1st agent ringing**
<img width="1678" alt="Screen Shot 2020-09-23 at 7 06 25 PM" src="https://user-images.githubusercontent.com/4672952/94092629-1aa0eb80-fdd0-11ea-8b1f-d641b394dd41.png">

**Figure 2: 2nd agent ringing**
<img width="1679" alt="Screen Shot 2020-09-23 at 7 06 33 PM" src="https://user-images.githubusercontent.com/4672952/94092643-2391bd00-fdd0-11ea-855a-c7105c23194c.png">

**Figure 3: Both clients are busy**
<img width="1678" alt="Screen Shot 2020-09-23 at 7 06 41 PM" src="https://user-images.githubusercontent.com/4672952/94092662-2d1b2500-fdd0-11ea-9eee-effaf0ccbc0e.png">
